Build failure: mpv-unwrapped

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Build mpv 0.40.0 on aarch64 darwin.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/293727780

### Relevant log output

```console
[215/259] Compiling Objective-C object libmpv.2.dylib.p/player_clipboard_clipboard-mac.m.o
FAILED: libmpv.2.dylib.p/player_clipboard_clipboard-mac.m.o 
clang -Ilibmpv.2.dylib.p -I. -I.. -I../include -Icommon -Ietc -Iplayer/javascript -Iplayer/lua -Isub -ITOOLS/osxbundle -Iosdep/mac -I/nix/store/ffc3cmyhz7i9wr9cxkxnrh8j5gm9cqm8-libass-0.17.3-dev/include -I/nix/store/shr80pw02b4bl6ivxd2b9594xk00hb6c-ffmpeg-7.1.1-dev/include -I/nix/store/0lm47qffzjgy28m1czbzdxglqhamrxby-libplacebo-7.349.0/include -I/nix/store/a82gq769kr6rczc4r8jr77lw6j9ibhsl-libdvdnav-6.1.1/include -I/nix/store/5f5i2apx6h47l4w5f0b9lcsnncs6c6y2-mujs-1.3.6/include -I/nix/store/9h9b8k15aqiarx7cs2za1hii8n9z7vfn-lcms2-2.17-dev/include -I/nix/store/jmj6b3xdvmd94b7kchng70k5rzgdi5yz-libarchive-3.7.7-dev/include -I/nix/store/5kp2jk7zx4gp8588y280lv0v1rj94cfw-libbluray-1.3.4/include -I/nix/store/14gghgzkfkaxv1md5jf371pzxwwd5ib6-lua-5.2.4/include -I/nix/store/ic2jpkxdxwlnffjdp9sfqcj199msg8kr-rubberband-4.0.0/include -I/nix/store/c1iz2dadkm7rg4lmq2r5adwsvrcfzzwm-uchardet-0.0.8-dev/include/uchardet -I/nix/store/nk1sr715rg8cawhnmbjwv809shmffyyc-zimg-3.0.5-dev/include -I/nix/store/2a716x8y7c8qa0jbnnsb60bz98173lr2-zlib-1.3.1-dev/include -I/nix/store/l9wzvcq6vlndkq3s1pf9r1nyvg4cwsxm-openal-soft-1.23.1/include -I/nix/store/l9wzvcq6vlndkq3s1pf9r1nyvg4cwsxm-openal-soft-1.23.1/include/AL -I/nix/store/x3n41q00m70308fmwpczpc4l70gbl74c-libcaca-0.99.beta20-dev/include -I/nix/store/2hi2frxa4wm9xhy4g2rbjdsqdkmrnj5w-vulkan-headers-1.4.304.0/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=c11 -D_FILE_OFFSET_BITS=64 -Wdisabled-optimization -Wempty-body -Wformat -Wimplicit-fallthrough -Wparentheses -Wpointer-arith -Wshadow -Wundef -Wvla -Wno-cast-function-type -Wno-format-zero-length -Wno-missing-field-initializers -Wno-sign-compare -Wno-switch -Wno-unused-parameter -fno-math-errno -fno-signed-zeros -fno-trapping-math -Werror=format-security -D_GNU_SOURCE -Wmissing-prototypes -Wstrict-prototypes -Werror=implicit-function-declaration -Wno-pointer-sign -MD -MQ libmpv.2.dylib.p/player_clipboard_clipboard-mac.m.o -MF libmpv.2.dylib.p/player_clipboard_clipboard-mac.m.o.d -o libmpv.2.dylib.p/player_clipboard_clipboard-mac.m.o -c ../player/clipboard/clipboard-mac.m
In file included from ../player/clipboard/clipboard-mac.m:19:
./osdep/mac/swift.h:270:9: fatal error: '/private/tmp/nix-build-mpv-0.40.0.drv-0/TemporaryDirectory.QMqOVt/app_bridge_objc-1.pch' file not found
#import "/private/tmp/nix-build-mpv-0.40.0.drv-0/TemporaryDirectory.QMqOVt/app_bridge_objc-1.pch"
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[216/259] Compiling Objective-C object libmpv.2.dylib.p/osdep_mac_app_bridge.m.o
FAILED: libmpv.2.dylib.p/osdep_mac_app_bridge.m.o 
clang -Ilibmpv.2.dylib.p -I. -I.. -I../include -Icommon -Ietc -Iplayer/javascript -Iplayer/lua -Isub -ITOOLS/osxbundle -Iosdep/mac -I/nix/store/ffc3cmyhz7i9wr9cxkxnrh8j5gm9cqm8-libass-0.17.3-dev/include -I/nix/store/shr80pw02b4bl6ivxd2b9594xk00hb6c-ffmpeg-7.1.1-dev/include -I/nix/store/0lm47qffzjgy28m1czbzdxglqhamrxby-libplacebo-7.349.0/include -I/nix/store/a82gq769kr6rczc4r8jr77lw6j9ibhsl-libdvdnav-6.1.1/include -I/nix/store/5f5i2apx6h47l4w5f0b9lcsnncs6c6y2-mujs-1.3.6/include -I/nix/store/9h9b8k15aqiarx7cs2za1hii8n9z7vfn-lcms2-2.17-dev/include -I/nix/store/jmj6b3xdvmd94b7kchng70k5rzgdi5yz-libarchive-3.7.7-dev/include -I/nix/store/5kp2jk7zx4gp8588y280lv0v1rj94cfw-libbluray-1.3.4/include -I/nix/store/14gghgzkfkaxv1md5jf371pzxwwd5ib6-lua-5.2.4/include -I/nix/store/ic2jpkxdxwlnffjdp9sfqcj199msg8kr-rubberband-4.0.0/include -I/nix/store/c1iz2dadkm7rg4lmq2r5adwsvrcfzzwm-uchardet-0.0.8-dev/include/uchardet -I/nix/store/nk1sr715rg8cawhnmbjwv809shmffyyc-zimg-3.0.5-dev/include -I/nix/store/2a716x8y7c8qa0jbnnsb60bz98173lr2-zlib-1.3.1-dev/include -I/nix/store/l9wzvcq6vlndkq3s1pf9r1nyvg4cwsxm-openal-soft-1.23.1/include -I/nix/store/l9wzvcq6vlndkq3s1pf9r1nyvg4cwsxm-openal-soft-1.23.1/include/AL -I/nix/store/x3n41q00m70308fmwpczpc4l70gbl74c-libcaca-0.99.beta20-dev/include -I/nix/store/2hi2frxa4wm9xhy4g2rbjdsqdkmrnj5w-vulkan-headers-1.4.304.0/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=c11 -D_FILE_OFFSET_BITS=64 -Wdisabled-optimization -Wempty-body -Wformat -Wimplicit-fallthrough -Wparentheses -Wpointer-arith -Wshadow -Wundef -Wvla -Wno-cast-function-type -Wno-format-zero-length -Wno-missing-field-initializers -Wno-sign-compare -Wno-switch -Wno-unused-parameter -fno-math-errno -fno-signed-zeros -fno-trapping-math -Werror=format-security -D_GNU_SOURCE -Wmissing-prototypes -Wstrict-prototypes -Werror=implicit-function-declaration -Wno-pointer-sign -MD -MQ libmpv.2.dylib.p/osdep_mac_app_bridge.m.o -MF libmpv.2.dylib.p/osdep_mac_app_bridge.m.o.d -o libmpv.2.dylib.p/osdep_mac_app_bridge.m.o -c ../osdep/mac/app_bridge.m
In file included from ../osdep/mac/app_bridge.m:23:
./osdep/mac/swift.h:270:9: fatal error: '/private/tmp/nix-build-mpv-0.40.0.drv-0/TemporaryDirectory.QMqOVt/app_bridge_objc-1.pch' file not found
#import "/private/tmp/nix-build-mpv-0.40.0.drv-0/TemporaryDirectory.QMqOVt/app_bridge_objc-1.pch"
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.3.0, macOS 15.3.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/nix/store/n5zhvsrwjrlyaimd3cp7s70j401iycx5-source`

### Notify maintainers

@fpletz 
@globin 
@Ma27

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/video/mpv/default.nix
+++ b/pkgs/applications/video/mpv/default.nix
@@ -227,6 +227,13 @@ stdenv.mkDerivation (finalAttrs: {
     ++ lib.optionals zimgSupport [ zimg ]
     ++ lib.optionals stdenv.hostPlatform.isLinux [ nv-codec-headers-11 ];
 
+  # https://github.com/mpv-player/mpv/issues/15591#issuecomment-2764797522
+  # In file included from ../player/clipboard/clipboard-mac.m:19:
+  # ./osdep/mac/swift.h:270:9: fatal error: '.../app_bridge_objc-1.pch' file not found
+  env = lib.optionalAttrs (stdenv.hostPlatform.isDarwin) {
+    NIX_SWIFTFLAGS_COMPILE = "-disable-bridging-pch";
+  };
+
   postBuild = lib.optionalString stdenv.hostPlatform.isDarwin ''
     pushd .. # Must be run from the source dir because it uses relative paths
     python3 TOOLS/osxbundle.py -s build/mpv

